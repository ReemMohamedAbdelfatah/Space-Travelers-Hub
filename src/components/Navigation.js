import { NavLink } from 'react-router-dom';
import logo from '../assets/logo.png';

const Navigation = () => (
  <nav className="navigation">
    <div className="logoDiv">
      <img src={logo} alt="logo..." className="logoImage" />
      <h2 className="logoText">Space Travelers&apos; Hub</h2>
    </div>
    <ul className="NavUl">
      <li className="navList">
        <NavLink
          to="/"
          className={({ isActive }) => (isActive ? 'active' : 'deActive')}
        />
      </li>
      <li className="navList">
        <NavLink
          to="rockets"
          className={({ isActive }) => (isActive ? 'active' : 'deActive')}
        >
          Rockets
        </NavLink>
      </li>

      <li className="navList">
        <NavLink
          to="missions"
          end
          className={({ isActive }) => (isActive ? 'active' : 'deActive')}
        >
          Missions
        </NavLink>
      </li>
      <li className="navList">
        <NavLink
          to="my-profile"
          className={({ isActive }) => (isActive ? 'active' : 'deActive')}
        >
          | My Profile
        </NavLink>
      </li>
    </ul>
  </nav>
);

export default Navigation;
