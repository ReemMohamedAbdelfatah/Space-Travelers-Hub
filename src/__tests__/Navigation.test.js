import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter as Router } from 'react-router-dom';
import Navigation from '../components/Navigation';

it('The Navigation should render correctly!', () => {
  const navigation = render(
    <Router>
      <Navigation />
    </Router>,
  );
  expect(navigation).toMatchSnapshot();
});
