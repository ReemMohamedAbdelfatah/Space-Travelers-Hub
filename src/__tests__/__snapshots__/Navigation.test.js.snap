// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Navigation should render correctly! 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        class="navigation"
      >
        <div
          class="logoDiv"
        >
          <img
            alt="logo..."
            class="logoImage"
            src="logo.png"
          />
          <h2
            class="logoText"
          >
            Space Travelers' Hub
          </h2>
        </div>
        <ul
          class="NavUl"
        >
          <li
            class="navList"
          >
            <a
              aria-current="page"
              class="active"
              href="/"
            />
          </li>
          <li
            class="navList"
          >
            <a
              class="deActive"
              href="/rockets"
            >
              Rockets
            </a>
          </li>
          <li
            class="navList"
          >
            <a
              class="deActive"
              href="/missions"
            >
              Missions
            </a>
          </li>
          <li
            class="navList"
          >
            <a
              class="deActive"
              href="/my-profile"
            >
              | My Profile
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav
      class="navigation"
    >
      <div
        class="logoDiv"
      >
        <img
          alt="logo..."
          class="logoImage"
          src="logo.png"
        />
        <h2
          class="logoText"
        >
          Space Travelers' Hub
        </h2>
      </div>
      <ul
        class="NavUl"
      >
        <li
          class="navList"
        >
          <a
            aria-current="page"
            class="active"
            href="/"
          />
        </li>
        <li
          class="navList"
        >
          <a
            class="deActive"
            href="/rockets"
          >
            Rockets
          </a>
        </li>
        <li
          class="navList"
        >
          <a
            class="deActive"
            href="/missions"
          >
            Missions
          </a>
        </li>
        <li
          class="navList"
        >
          <a
            class="deActive"
            href="/my-profile"
          >
            | My Profile
          </a>
        </li>
      </ul>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
